go/vt/vttablet/tabletmanager/restore.go

```diff
@@ -117,6 +114,21 @@ func (tm *TabletManager) restoreDataLocked(ctx context.Context, logger logutil.L
 		StartTime:           logutil.ProtoToTime(keyspaceInfo.SnapshotTime),
 	}
 
+	ok, err := mysqlctl.ShouldRestore(ctx, params)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		params.Logger.Infof("Attempting to restore, but mysqld already contains data. Assuming vttablet was just restarted.")
+		return mysqlctl.PopulateMetadataTables(params.Mysqld, params.LocalMetadata, params.DbName)
+	}
+	// should not become master after restore
```

Comment:
```suggestion
	// We should not become master after restore, because that would falsely
	// start a new master term, and it's likely our data dir will be out of date.
```